### PR TITLE
Diurnal [WIP]

### PR DESCRIPTION
**Pull request by doutriaux1**
Still need to add test suite

### CODE REVIEW

**Comment by doutriaux1** [comment]
@durack1 @gleckler1 this is ready to be approved and go in

**Comment by durack1** [comment]
@doutriaux1 should we wait for a green light with the tests? Or do we still have the precision issue causing the red crosses?

**Comment by doutriaux1** [comment]
It's green now.

File: share/default_regions.py [hunk 1]
```diff
@@ -1,44 +1,44 @@
 import cdutil
 
 regions_specs = {
-    'NHEX': {'domain': cdutil.region.domain(latitude=(30., 90, 'ccb'))},
```
Comment (durack1) [inline share/default_regions.py]: @doutriaux1 why was this truncated?

File: src/python/diurnal/common.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+P.add_argument("-t", "--filename_template",
+               default="cmip5.%(model).%(experiment).%(realization).%(frequency).%(realm).%(frequency).%(variable).%(version).latestX.xml")  # noqa
+P.add_argument("--skip", default=[],
+               help="models to skip", nargs="*")
+P.add_argument(
+    "-a",
+    "--append",
+    default=False,
+    action="store_true",
+    help="append in json file in json exist (e.g. adding a model to file)")
```
Comment (durack1) [inline src/python/diurnal/common.py]: "append in json file in json exist (e.g. adding a model to file)" -> "append in json file if json exists (e.g. adding a model to an existing file)"

File: src/python/diurnal/fourierFFT.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
+﻿def fastFT(x, t):
+    '''
+Use a Numerical Python function to compute a FAST Fourier transform -- which should give the same result as a simple
+SLOW Fourier integration via the trapezoidal rule.
+
+Return mean + amplitudes and times-of-maximum of the first three Fourier harmonic components of a time series x(t).
+Do NOT detrend the time series first, in order to retain the "sawtooth" frequency implied by the input length of the
+time series (e.g. the 24-hour period from a composite-diurnal cycle).
+
+On input: x[i, j] = values      at each gridpoint (i) for N times (j), e.g. N = 8 for a 3-hr composite-diurnal cycle
```
Comment (durack1) [inline src/python/diurnal/fourierFFT.py]: Extra spaces in doc string

File: src/python/diurnal/fourierFFT.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+This version of fastFT (see above) does all gridpoints at once.
+
+Use a Numerical Python function to compute a FAST Fourier transform -- which should give the same result as a simple
+SLOW Fourier integration via the trapezoidal rule.
+
+Return mean + amplitudes and times-of-maximum of the first three Fourier harmonic components of a time series x(t).
+Do NOT detrend the time series first, in order to retain the "sawtooth" frequency implied by the input length of the
+time series (e.g. the 24-hour period from a composite-diurnal cycle).
+
+On input: x[k,i,j] = values      at each gridpoint (i,j) for N times (k), e.g. N = 8 for a 3-hr composite-diurnal cycle
```
Comment (durack1) [inline src/python/diurnal/fourierFFT.py]: Same here - spaces

File: share/default_regions.py (hunk 1 above)
Comment (doutriaux1) [reply share/default_regions.py]: No idea. Let me put it back in. Nice catch.

**Pull request merged**